### ISSUE TITLE
Where are things headed?

### ISSUE THREAD

**Issue by jachin84** [robladbrook/slack#21]
Hi,
I noticed your slack project while browsing Github the other day and really like how clean your project looks. I'm considering building up a Windows Service and building a bot using your library. I'm just wondering where you intend on taking the project and if you intend on working on the project on a regular basis?

I potentially have time I can spend on contributing as I'm building a bot for my company. I'm not a seasoned dev, more of an ops guy but know enough to be helpful.

**Comment by robladbrook** [comment]
Hi Jacob

A windows service wrapper is definitely on the roadmap for the project, so it would be great to see what you come up with!

We're using Slack extensively so this project will continue to evolve for the foreseeable future and will likely add more developers to it. I'm hoping to do a proper release on Nuget within the next 4 weeks to bring it out of alpha.

Any ideas/feedback/suggestions etc you have just throw them my way.

Cheers

**Comment by jachin84** [comment]
Yeah great. I'll create a fork and start work on a service perhaps later in the week. I'll do a PR when its ready so you can take a look. Some things that are definitely important to me:
- Authorisation - we need the ability to limit who can run certain commands. A lot of project implement this in a similar way to hubot but I'd like to see this as a first-class citizen managed centrally. 
- Webhooks - it would be great to expose some webhooks for services to call and the bot to take action upon.
- PowerShell support - on the windows side many IT Pros can do amazing things with PowerShell. It would be good to have an extension that allows handlers written in PowerShell. There is another project here on GitHub that does this called [Mmbot](https://github.com/mmbot/mmbot/).
- Configuration - Most people on the Windows side would be familiar enough with .config files for runtime configuration. I'd be interested in your thoughts here.
- Azure support - Support for deploying to Azure using ARM template.
- Easy - It's gotta be easy. I like the idea of a dedicated slack bot not a generic chat bot. I think it makes the implementation much simpler and will aid in creating a first-class bot.
